weldx/time.py

```diff
@@ -0,0 +1,125 @@
+"""Contains classes and functions related to time."""
+
+
+from typing import TYPE_CHECKING, List, Union
+
+if TYPE_CHECKING:
+    from pandas import DatetimeIndex, TimedeltaIndex, Timestamp
+    from pint import Quantity
+    from xarray import DataArray
+
+    from weldx.types import types_time_like, types_timestamp_like
+
+
+class Time:
+    """Provides a unified interface for time related operations."""
+
+    def __init__(
```

Comment:
I think one of the core question to decide is what datatype(s) to use for storing the actual data inside the class

I would suggest using the basic `pandas` types here (`Timedelta`, `TimedeltaIndex`, `Timestamp`, `DateTimeIndex`) and storing them in a simple unified place like `_data` (original, i know 😉)
Every input gets converted into the appropriate pandas type

That way, most of the actual operations can be implemented using basic pandas syntax which should be very easy to implemented (and also throw reasonable exceptions for most cases without us having to worry about them).
I also expect the performance to be superior to using `xarray`.

We should also discuss and define some very basic implementation details of the class:
- do we allow duplicate values (I'd say no)
- do we enforce the time values to be sorted (I'd say yes!)


I added this comment to the `__init__` function so that we can mark it as 'resolved' later

Reply:
The docstring of the `__init__` methods' `time` parameter already agrees with your assessment of the two bullet points ("ascending order"). Since I wrote it I guess it is settled then 😉 

Regarding the pandas time types: This is probably the way that makes the most sense performance-wise and in every other context. 

I think we should also review the usage of the special xarray time structure in other classes. Most of the use cases are related to having a unified interface, that will now be provided by this class. But this is something for later, if the class itself is finished

Reply:
I added a check for monotonic increasing values after conversion to the equivalent pandas types

that should be safe enough since we don't do any kind of sorting beforehand